notebooks/ch17/classification_multiclass.md

```diff
@@ -0,0 +1,312 @@
+    
+    sns.lmplot('$x_1$', '$x_2$', data=data_copy, hue='$y$ == ' + label, hue_order=[True, False], 
+               markers=[markers[label][0], 'x'], palette=[markers[label][1], 'gray'],
+               fit_reg=False)
+    plt.xlim(1.0, 4.0)
+    plt.ylim(1.0, 4.0);
+```
+
+### (outline)
+
+- What is multiclass classification
+    - More than 2 labels that can be classified; versus a binary classification which is restricted to 2 labels
+- A situation where multiclass classification might be used
+    - Preferably come up with an example in lebron.csv, but might have to use a different data set
+- OvO vs OvR
+    - In logistic regression, multiclass classification comes down to a bunch of binary problems
+    - OvO: Fit a binary classification problem to each pair of classes, then vote for class that wins the most "head-to-head" classifications
+    - OvR/OvA: Fit a binary classification problem for each class versus the rest of the classes; result is class that has the highest probability
+- Example walking through multiclass? sklearn's logistic regression supports OvR out of the box.
+- Multilabel: More than one classification problem (e.g. a document can be positive/negative, religion/not religion, political/apolitical, etc.)
+
+## Multiclass Classification
+
+So far we have discussed binary classification, in which our classifier determines whether an observation is part of the positive class (e.g. an email is classified as spam) or the negative class (e.g. an email is not spam). However, many data science problems involve **multiclass classification**, in which we would like to classify observations as one of several different classes. For example, we may be interested in classifying emails into folders such as Family, Friends, Work, and Promotions. To solve these types of problems, we use a new method called **one-vs-rest (OvR) classification**.
+
+### One-Vs-Rest Classification
+
+In OvR classification (also known as one-vs-all, or OvA), we decompose a multiclass classification problem into several different binary classification problems. For example, we might observe training data as shown below:
+
+
+```python
+# HIDDEN
+shapes = pd.DataFrame(
+    [[1.3, 3.6, 'triangle'], [1.6, 3.2, 'triangle'], [1.8, 3.8, 'triangle'],
+     [2.0, 1.2, 'square'], [2.2, 1.9, 'square'], [2.6, 1.4, 'square'],
+     [3.2, 2.9, 'circle'], [3.5, 2.2, 'circle'], [3.9, 2.5, 'circle']],
+    columns=['$x_1$', '$x_2$', '$y$']
+)
+```
+
+
+```python
+sns.lmplot('$x_1$', '$x_2$', data=shapes, hue='$y$', markers=['^', 's', 'o'], fit_reg=False)
+plt.xlim(1.0, 4.0)
+plt.ylim(1.0, 4.0);
+```
+
+
+![png](classification_multiclass_files/classification_multiclass_5_0.png)
+
+
+Our goal is to build a multiclass classifier that labels observations as `triangle`, `square`, or `circle` given values for $x_1$ and $x_2$. First, we want to build a binary classifier that predicts observations as `triangle`/not `triangle`:
```

Comment:
> we want to build a binary classifier

I think you should mention the name of your classifier `lr_triangle` here (and likewise for `lr_square` and `lr_circle` with their graphs) since we can see the graph and the `plot_binary` function but it isn't clear that an actual LogisticRegression model object is being created.